### PR TITLE
Allow varying atom counts, fix VS Code worker stall, speed up XYZ/LAMMPS parsing

### LINKED ISSUE #449: Trajectory loading forces consistent atom count across trajectory
### Bug Report

Hi Janosh,

The most recent refactors introduced the limitation that trajectories fail to load if frames do not contain the same number of atoms. This behavior was one of the strong features for me because I could dump generated structures into a single XYZ file and scroll through them easily. 

I wanted to check if you have thoughts on that before I make a PR to restore that flexibility.

### LINKED ISSUE #451: VS Code webviews: worker spawn hangs ~20 s per file open (WorkerUnavailableError: Failed to fetch dynamically imported module)
### Bug Report

## Summary                                                                                                                                                                                                                                       

 With the VS Code extension (v0.6.0, Windows), every file open shows a blank webview for ~20 seconds before anything renders. The console always logs:

 ```
   parse_in_worker: no worker for <file>, parsing on the main thread:
   WorkerUnavailableError: Uncaught TypeError: Failed to fetch dynamically imported module:
   https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/<me>/.vscode/extensions/janosh.matterviz-0.6.0/dist/assets/parse-worker-*.js
 ```

 Parsing then succeeds on the main thread, so this is purely a fixed ~20 s tax on every interaction with the viewer.

## Root cause

 Webview assets are served through an internal service worker that only intercepts document resource loads. A Worker script request runs outside that pipeline:

 1. The browser tries to resolve https://file+.vscode-resource.vscode-cdn.net over real DNS, which is not a real host. Evidence from the webview console:
     - fetch(workerUrl) → net::ERR_NAME_NOT_RESOLVED in ~1 ms
     - Network tab: parse-worker-*.js sits (pending) for the whole stall while all regular assets load in milliseconds via (ServiceWorker)
 2. new Worker() additionally requires a TrustedScriptURL under VS Code's Trusted Types enforcement — raw string URLs are rejected instantly (This document requires 'TrustedScriptURL' assignment).

 The blob-module shim added in #438 gets past (2) — the blob itself constructs fine — but its inner await import(<vscode-resource URL>) runs in worker context and hits (1), which is where the hang comes from. Pre-#438 versions failed the
 same way instantly and silently (no workers in webviews at all); #438 unintentionally turned that into a visible 20-second stall per open.

## Possible fixes

 - Ship workers as self-contained single files and load them via fetch(url) → blob → URL.createObjectURL → new Worker(blobUrl) (Microsoft's documented pattern). Fetching happens on the main thread, so it goes through the service worker
   and works. Note the current ES-format code splitting has to become single-file for this — a blob module can't resolve relative chunk imports.
 - Register a narrow Trusted Types policy for the blob URLs regardless of approach, since VS Code enforces TT on Worker constructors.
 - Or simply skip workers when running inside a webview (location.protocol === 'vscode-webview:') and always parse on the main thread, as pre-#438 did — instant, at the cost of off-thread parsing.

### PR DESCRIPTION
**Pull request by janosh**
Closes #449
Closes #451

## Varying atom counts (#449)

The per-frame viewer never needed a constant atom count. Only `collect_positions` (MSD/VACF/RDF/trails) does, and `PositionAccumulator.add_frame` already rejects such runs there with a precise message. Two places enforced that analysis-only invariant at load time and broke the "bag of generated structures in one XYZ" workflow.

- `runs/memory.ts` `validate_frames`: drops the cross-frame atom-count check; each frame must still have non-empty sites with finite coordinates. Masses and per-atom signals still validate against frame 0. Affects every eager parser (XYZ, JSON, pymatgen, ASE, VASP).
- `parse/lammps.ts`: frames whose atom-ID set changed (GCMC, deposition) are kept, sorted by ID, instead of being dropped with a warning. A multi-frame dump without an `id` column loads with a warning instead of throwing.
- Analyses on a variable-count run fail at `collect_positions` with `Atom count changed at frame N: expected X atoms, got Y`; frame scrubbing is unaffected.

## VS Code webview ~20 s stall on every file (#451)

- `file-viewer/main.ts`: the blob-module shim from #438 let `new Worker` succeed, but its inner `import()` of the vscode-cdn resource URL runs in worker context where the webview service worker does not intercept, so Chromium resolved the fake host over DNS and hung ~20 s before the main-thread fallback kicked in. Cross-origin worker scripts now throw a `SecurityError` synchronously; every worker client (parse, MSD, VACF, RDF, structure-id, isosurface) already falls back to main-thread compute on a constructor throw, so the fallback is instant. Real off-thread work in a webview needs self-contained single-file worker bundles fetched on the main thread and handed over as blob URLs under a Trusted Types policy; the comment records that path.
- `parse-in-worker.ts`: indexable trajectories (multi-frame XYZ, ASE `.traj`) are exempt from the 25 MiB main-thread fallback ceiling. Above the indexing threshold they are located frame by frame without decoding, which is what the host's large-file path already does on the main thread; with webviews always on the main thread the ceiling would have made 25–400 MB dumps unopenable there.

## Parsing performance (XYZ, LAMMPS)

Profiled with `--cpu-prof` on a 200 frames × 2000 atoms EXTXYZ (20 MB) and a 28 MB LAMMPS dump. Baseline hot spots: `trim().split(/\s+/)` + `Number()` per column, the regex line split run twice (sniff + parse), `Math.hypot(...spread)` per atom, a second force-stats pass in the property extractor, and GC from one string object per line.

- `structure/parsers/shared.ts` `LineScanner`: tokenizes a `(text, from, to)` range on char codes and decodes plain decimals in the same pass as an exact integer mantissa scaled by one exact power of ten, which rounds bit-identically to `Number()` (exponents, Fortran `D`, long mantissas take the `parse_float_token` slow path). Token bounds and values live in reusable typed arrays, so nothing is allocated per coordinate. A test compares ~4000 generated and edge-case tokens bit for bit against `parse_float_token`.
- `trajectory/helpers.ts`: `iter_xyz_frames` / `index_xyz_frames` / `build_xyz_frame` walk the untouched text by `indexOf` offsets instead of `split(/\r?\n/)`. A 10M-line dump no longer pays a full copy pass and an array of 10M string objects that outweighs the text itself; indexed runs retain only the text plus frame offsets. Frame specs carry count-line offset, 1-based line number, atom-block offsets and comment; error messages keep their frame/line numbering. The structure XYZ reader and NEB path reader share the offsets.
- `has_multiple_xyz_frames` sniffs a 64 KB head first and only escalates to 8 MB / the full text when the cut tore a frame, so format detection no longer splits the whole file.
- `split_lines` (LAMMPS, XDATCAR) only runs the `\r?\n` regex when the text contains `\r`.
- `force_stress_data_extractor` reuses the parser's `force_max` / `force_norm` instead of a second pass over every atom; `calc_force_stats` drops the spread call; `get_density` sums masses in one pass; `validate_frames` uses indexed loops.

Same-session numbers (machine under heavy external load, so treat as relative): `open_trajectory` eager end to end 695 → 310 ms, indexed frame decode 2.5 → 0.9 ms per 2000-atom frame, LAMMPS parse 613 → 450 ms, indexed open of the 20 MB file ~10 ms.

### CODE REVIEW

File: src/lib/trajectory/parse/lammps.ts [hunk 1]
```diff
@@ -319,21 +318,6 @@ export function parse_lammps_trajectory(
       ).toSorted(
         (left_idx, right_idx) => numeric_atom_ids[left_idx] - numeric_atom_ids[right_idx],
       )
-      const sorted_atom_ids = order.map((atom_idx) => numeric_atom_ids[atom_idx])
-      const expected_atom_ids = reference_atom_ids
```
[the remark this answers is not in the document]
Comment (janosh) [reply src/lib/trajectory/parse/lammps.ts:323]: Fixed in 67d0844: PositionAccumulator now records each site's `id` property from the first frame and rejects a later frame whose site carries a different ID, so an equal-count atom swap fails at collect_positions (`Atom identity changed at frame N…`) while the frames stay browsable. Regression test: [1,2] → [1,3] in the GCMC parser test.
